File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int> valuesToRemove(nums.begin(),nums.end());
        while(head!=NULL && valuesToRemove.count(head->val)==1){
            head= head->next;
        }
        if(head!=NULL){
            ListNode* temp= head;
            while(temp->next){
                if(valuesToRemove.count(temp->next->val)==1){
                    temp->next= temp->next->next;
                }
                else temp= temp->next;
            }
        }
        return head;
    }
};

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        HashSet<Integer> valuesToRemoveFromList= new HashSet<>();
        for(int i=0;i<nums.length;i++){
            valuesToRemoveFromList.add(nums[i]);
        }
        while(head!=null && valuesToRemoveFromList.contains(head.val)){
            head= head.next;
        }
        ListNode temp= head;
        if(head!=null){
            while(temp.next!=null){
                if(valuesToRemoveFromList.contains(temp.next.val)){
                    temp.next= temp.next.next;
                }
                else{
                    temp= temp.next;
                }
            }
        }
        return head;
    }
    
}